### DISCUSSION TITLE
extracting solution from a binvar

### DISCUSSION THREAD

**Discussion by zoharl3** [yalmip/YALMIP#1065] [General discussion] [answered]
So far, I've been extracting a solution like this:

`x = double(x)`

I'm looking at the tutorial, and it suggests to use value(). I didn't follow the source, but it seems to produce the same result.

Anyway, it might be obvious, but I spent two hours to realize (find the bug) that if I have

`b = binvar(1)`

then

`b = logical( double(b) )`

may be incorrect, and I should do:

`b = logical( round( double(b) ) )`

**Comment by johanlofberg** [comment, marked as answer]
double is obsolete, the command is called value since many years back

**Reply by zoharl3** [reply]
Okay, I'm still not sure what's the difference, but my point was that binvar and intvar are still an sdpvar wrapping a double and should be treated as such (i.e., they aren't int).

**Reply by johanlofberg** [reply]
Obsoleted @sdpvar/double and new @sdpvar/value does exactly the same, but @sdpvar/double is to be removed as it sends the wrong signal about doing something different from what it does. Built.in double is about casting numbers and converting number format, while @sdpvar/value really only is about extracting the computed solution, and the computed solution is always a floating-point number (i.e. a double) regardless, you have only told the solver to search for (near-)integer solutions..

If you for some reason want to cast the computed solution to some other format, such as short, int, logical or what ever, that is outside the whole solver/yalmip framework
